pkg/config/validator.go

```diff
@@ -302,20 +303,20 @@ func (p *validator) validateAdapters(key string, cfg string) (ce *adapter.Config
 }
 
 // ValidateSelector ensures that the selector is valid per expression language.
-func (p *validator) validateSelector(selector string) (err error) {
+func (p *validator) validateSelector(selector string, df expr.AttributeDescriptorFinder) (err error) {
 	// empty selector always selects
 	if len(selector) == 0 {
 		return nil
 	}
-	return p.exprValidator.Validate(selector)
+	return p.exprValidator.AssertType(selector, df, dpb.BOOL)
```

Comment:
Is this safe ? I think it is since selectors are always bool type. This will allow us to get rid of the exprValidator.Validate method completely as it only checks the syntax and not the type.